api/client/event/event_stream.go

```diff
@@ -93,6 +93,8 @@ func (h *EventStream) Subscribe(eventsChannel chan<- *Event) {
 			EventType: EventConnectionError,
 			Data:      []byte(errors.Wrap(err, client.ErrConnectionIssue.Error()).Error()),
 		}
+		close(eventsChannel)
```

Comment:
What is the panic? Can you provide a regression test to ensure we maintain this fix?

Reply:
sure, I can add a test. The panic occurs in the defer when calling resp.Body.Close(). Depending on timing, you may or may not get the error event before it panics. Even without the defer, or checking for the nil pointer, seems the error should still return since there will never be any events sent over.

```
event.Type connection_error
event.Data could not connect: Get "MYBADHOST/eth/v1/events?topics=head": net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x00\x00\x06\x04\x00\x00\x00\x00\x00\x00\x05\x00\x00@\x00"
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104025c0c]

goroutine 226 [running]:
github.com/prysmaticlabs/prysm/v5/api/client/event.(*EventStream).Subscribe.func1()
        /Users/csweeney/go/pkg/mod/github.com/prysmaticlabs/prysm/v5@v5.1.2/api/client/event/event_stream.go:99 +0x2c
panic({0x104a39a40?, 0x105bc29b0?})
        /opt/homebrew/Cellar/go/1.23.2/libexec/src/runtime/panic.go:785 +0x124
github.com/prysmaticlabs/prysm/v5/api/client/event.(*EventStream).Subscribe(0xc000f558c0, 0xc000657c00)
        /Users/csweeney/go/pkg/mod/github.com/prysmaticlabs/prysm/v5@v5.1.2/api/client/event/event_stream.go:104 +0x578
```